src/rules/stylelint-disable-reason/index.js

```diff
@@ -0,0 +1,59 @@
+import {
+  report,
+  ruleMessages,
+  validateOptions,
+} from "../../utils"
+
+export const ruleName = "stylelint-disable-reason"
+
+export const messages = ruleMessages(ruleName, {
+  expectedPreceding: "Expected comment reason before stylelint disable comment",
+  expectedSucceeding: "Expected comment reason after stylelint disable comment",
+})
```

Comment:
I would prefer the simpler "before" and "after" pair to "preceding" and "succeeding".